algorithm.go

```diff
@@ -164,3 +173,20 @@ func (a Algorithm) FromBytes(p []byte) Digest {
 func (a Algorithm) FromString(s string) Digest {
 	return a.FromBytes([]byte(s))
 }
+
+// Validate validates the encoded portion string
+func (a Algorithm) Validate(encoded string) error {
+	r, ok := anchoredEncodedRegexps[a]
+	if !ok {
+		return ErrDigestUnsupported
+	}
+	// Digests much always be hex-encoded, ensuring that their hex portion will
+	// always be size*2
+	if a.Size()*2 != len(encoded) {
```

Comment:
We can drop this block.  For algorithms in `anchoredEncodedRegexps` we're already checking the length (via the anchored regexp).  And I'm not sure what would happen here if the user failed to import `crypto/sha512` or similar.  Would `Algorithm("sha512:abc…")` return zero and then erroneously trigger this condition?  Better to drop the redundant check.